### PR TITLE
add meson commands to the Makefile

### PR DESCRIPTION
**Pull request by rhatdan**
I am not familiar with meson, so I think this will make it easier to discover the commands with a Makefile workflow.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Review by mwperina — COMMENTED** [review]
I'm not sure about this, our goal was to use meson for everything, so remaining targets in Makefile has been left there only because we don't have yet solution in meson, which would do the same thing with the same comfort

**Comment by rhatdan** [comment]
I think you should enable both workflows.

**Comment by engelmi** [comment]
> I'm not sure about this, our goal was to use meson for everything, so remaining targets in Makefile has been left there only because we don't have yet solution in meson, which would do the same thing with the same comfort

You are right, Martin. Although after using meson now for a while and always having to type the same command patterns, I think having make targets to combine those improves the flow. And I agree with Dan that having the commands in a Makefile would help getting started with meson and the project.

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
+DESTDIR ?=
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+all:
+	meson setup builddir
+	meson compile -C builddir
+
+test: all
+	meson setup builddir
```
Comment (engelmi) [inline Makefile]: `setup` and `compile` can be removed from the `test` target since it is calling `all`.

File: Makefile [hunk 2]
```diff
@@ -1,17 +1,33 @@
+.PHONY: all fmt check-fmt lint lint-fix codespell all
+
+DESTDIR ?=
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+all:
```
Comment (engelmi) [inline Makefile]: Minor: Maybe this can be renamed to `compile`?

File: Makefile [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
+	meson compile -C builddir
+
+test: all
+	meson setup builddir
+	meson compile -C builddir
+	meson test -C builddir
+
+install: all
+	meson install -C builddir --destdir "$(DESTDIR)"
+
```
Comment (engelmi) [inline Makefile:17]: What do you think of a `clean` target? For example, like this
```bash
rm -rf builddir
meson setup builddir
```

Comment (rhatdan) [reply Makefile:17]: Yes definitely

File: Makefile [hunk 4]
```diff
@@ -1,17 +1,38 @@
...
-lint-fix: 
+lint-fix:
 	eval $(ALL_SRC_FILES) | xargs -0 -I {} clang-tidy --quiet {} --fix -- -I src/ -D_GNU_SOURCE
 
 codespell:
 	codespell -S Makefile,imgtype,copy,AUTHORS,bin,.git,CHANGELOG.md,changelog.txt,.cirrus.yml,"*.xz,*.gz,*.tar,*.tgz,*ico,*.png,*.1,*.5,*.orig,*.rej,*.xml" -L keypair,flate,uint,iff,od,ERRO -w
+
+clean:
+	rm -rf builddir
+	meson setup builddir
```
Comment (ygalblum) [inline Makefile]: Why do you need to run the `setup` command during `clean`? Won't it run for `build`?

File: Makefile (hunk 3 above)
Comment (eli-schwartz) [reply Makefile:17]: Ninja has a clean target -- it cleans all built outputs, but doesn't remove the build directory entirely, so you still have a ninja file. Adding a Makefile target that purges the entire build directory seems confusingly different from what meson/ninja does -- and it's also different from what other build systems do.

e.g. autotools distinguishes between `make clean` (removes built outputs) and `make distclean` (removes generated builddir, Makefiles, etc.)

File: Makefile (hunk 4 above)
Comment (Yarboa) [reply Makefile]: It is a chicken and egg, since builddir already deleted, 
Maybe using this 
`meson setup --wipe builddir`

Could be better,

**Review by engelmi — APPROVED** [review]
LGTM

**Review by mwperina — APPROVED** [review]
LGTM

**Pull request merged**